src/bottompanel/compilewindow.cpp

```diff
@@ -0,0 +1,81 @@
+#include "compilewindow.h"
+#include "ui_compilewindow.h"
+#include "compiler/compilercontroler.h"
+#include <QGridLayout>
+#include <QListWidgetItem>
+#include <QDebug>
+
+CompileWindow::CompileWindow(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::CompileWindow)
+{
+    ui->setupUi(this);
+    compileOutputList = new QListWidget;
+    compileOutputList->setLayoutMode(QListView::Batched);
+
+    QGridLayout *compileOutputLayout = new QGridLayout(this);
+    compileOutputLayout->addWidget(compileOutputList);
+    this->setLayout(compileOutputLayout);
+}
+
+CompileWindow::~CompileWindow()
+{
+    delete ui;
+}
+
+void CompileWindow::setCompileOutput(QString text)
+{
+    compileOutputList->clear();
+    auto allErrorslist = getAllErrorsFromCompileOutput(text);
+    for (auto error : allErrorslist)
+    {
+        auto *errorItem = new QListWidgetItem(error);
+        if (error.contains(": warning: "))//if its warnings
```

Comment:
Use one comment style.

Reply:
I use one comment style. The code abowe are Volodymyrs comments